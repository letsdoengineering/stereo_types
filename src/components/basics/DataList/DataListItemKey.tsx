import React from 'react'
import getClassNames from '../../../utils/get-class-names'
import dataListStyles from './DataList.module.css'

type DataListItemKeyProps = React.AllHTMLAttributes<HTMLElement> & {
  /** optional: class-names */
  className?: string
  /** children always of type ReactNode */
  children: React.ReactNode
}

/** DataListItemKey: renders a DT tag */
const DataListItemKey: React.FC<DataListItemKeyProps> = ({
  className,
  children,
}: DataListItemKeyProps) => {
  const classNames = getClassNames({
    defaultClasses: [`${dataListStyles.itemKey}`],
    className,
  })

  return <dt className={classNames}>{children}</dt>
}

export default DataListItemKey
