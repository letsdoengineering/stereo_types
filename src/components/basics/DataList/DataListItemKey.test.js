import * as React from 'react'
import { render, waitFor, screen } from '@testing-library/react'
import DataListItemKey from './DataListItemKey'
import renderer from 'react-test-renderer'
import { axe } from 'jest-axe'

describe('DataListItemKey', () => {
  it('renders li', async () => {
    render(<DataListItemKey>a key</DataListItemKey>)
    const list = screen.getByText('a key')

    await waitFor(() => expect(list.nodeName).toBe('DT'))
  })

  it('has not changed the JSX layout of the page from previous snapshot', () => {
    const tree = renderer.create(<DataListItemKey>a key</DataListItemKey>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the page DOM that is accessible', async () => {
    const { container } = render(<DataListItemKey>a key</DataListItemKey>)
    /** dLitem rule ignored as this component on its own does not sit inside a <dl> */
    const results = await axe(container, {
      rules: {
        dlitem: { enabled: false },
      },
    })
    expect(results).toHaveNoViolations()
  })
})
