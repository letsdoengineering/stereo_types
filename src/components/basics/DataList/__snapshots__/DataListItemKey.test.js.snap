// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataListItemKey has not changed the JSX layout of the page from previous snapshot 1`] = `
<dt
  className="itemKey"
>
  a key
</dt>
`;
